File: src/main/java/com/senderman/stickersorterbot/bot/SorterBot.java
package com.senderman.stickersorterbot.bot;

import com.annimon.tgbotsmodule.BotHandler;
import com.annimon.tgbotsmodule.BotModule;
import com.annimon.tgbotsmodule.beans.Config;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

@Component
public class SorterBot implements BotModule {

    private final BotHandler botHandler;

    public SorterBot(BotHandler botHandler) {
        this.botHandler = botHandler;
    }

    @Override
    public @NotNull BotHandler botHandler(@NotNull Config config) {
        return botHandler;
    }
}

File: src/main/kotlin/com/senderman/stickersorterbot/bot/SorterBot.kt
package com.senderman.stickersorterbot.bot

import com.annimon.tgbotsmodule.BotHandler
import com.annimon.tgbotsmodule.BotModule
import com.annimon.tgbotsmodule.beans.Config
import org.springframework.stereotype.Component

@Component
class SorterBot(
        private val sorterBotHandler: SorterBotHandler
) : BotModule {

    override fun botHandler(config: Config): BotHandler = sorterBotHandler

}